Add docker

Kazantsevyury opened the pull request.

Kazantsevyury committed:
feat(gateway): Implement ShareItGateway for service routing

Divide the ShareIt application into two separate parts: ShareItGateway and the core service. The ShareItGateway handles incoming requests and routes them to the appropriate service, providing an abstraction layer for service interaction.

Kazantsevyury committed:
feat(server): Implement core services for ShareIt application

The ShareIt application has been structured into two distinct components. This commit introduces the core services in the ShareItServer part, providing the primary business logic and data management functionalities.

Kazantsevyury committed:
add pom.xml , LogginAspect

Kazantsevyury committed:
cheskstyle

Kazantsevyury committed:
cheskstyle

VladimirTitov4 commented on server/src/main/java/ru/practicum/shareit/booking/storage/BookingStorage.java:
```diff
@@ -25,45 +25,35 @@ public interface BookingStorage extends JpaRepository<Booking, Long> {
 
-    @Query(value = "SELECT b FROM Booking b JOIN FETCH b.item i JOIN FETCH b.booker u WHERE b.booker.id = ?1 AND b.start >= ?2 ORDER BY b.start DESC",
-            countQuery = "SELECT COUNT(b) FROM Booking b JOIN b.item i JOIN b.booker u WHERE b.booker.id = ?1 AND b.start >= ?2 ORDER BY b.start DESC")
+    @Query(value = "SELECT b FROM Booking b JOIN FETCH b.item i JOIN FETCH b.booker u WHERE b.booker.id = ?1 AND b.start >= ?2 ORDER BY b.start DESC")
```
Используйте :namedParameters, они нагляднее

Kazantsevyury committed:
refactor: use named parameters in JPA queries for clarity

- Update ItemStorage, BookingStorage, CommentStorage, and ItemRequestStorage interfaces
- Replace positional parameters with named parameters in all JPA queries
- Simplify count query in ItemRequestStorage for better performance
- Ensure consistency and readability across different repository interfaces

These changes improve maintainability and prevent potential errors associated with positional parameter misalignment in complex JPQL queries.
